gnd/src/main/java/com/google/android/gnd/ui/home/mapcontainer/MapContainerViewModel.java

```diff
@@ -397,6 +508,23 @@ public void onAddFeatureBtnClick() {
     addFeatureButtonClicks.onNext(getCameraPosition().getValue().getTarget());
   }
 
+  public void onAddPolygonBtnClick() {
+    if (vertices.size() >= 3) {
+      checkPointNearVertex(getCameraPosition().getValue());
+    } else {
+      vertices.add(getCameraPosition().getValue().getTarget());
+    }
+    updateDrawnPolygonFeature(ImmutableList.copyOf(vertices));
+  }
+
+  public void onConfirmButtonClick() {
+    confirmButtonClicks.onNext(getCameraPosition().getValue().getTarget());
+  }
+
+  public void onCancelButtonClick() {
+    cancelButtonClicks.onNext(Nil.NIL);
+  }
```

Comment:
Possible merge conflict. This was removed in https://github.com/google/ground-android/pull/970/files

Reply:
removed.